modules/manage/pages/raft-group-reconfiguration.adoc

```diff
@@ -0,0 +1,66 @@
+
+== About Raft group reconfiguration
+
+A Raft group’s member set changes during operations like cluster resize or xref:manage:cluster-maintenance/continuous-data-balancing.adoc[Continuous Data Balancing]. This change, also referred to as Raft group reconfiguration, includes adding, removing, or replacing Raft group members. When the group configuration changes, the current leader replicates a new configuration message and sends it to the followers. 
+
+Raft uses the joint consensus algorithm to maintain consistency during reconfiguration. It involves both the old member set and the proposed new member set reaching an agreement between them, either on accepting writes, or electing a new leader. The new leader can be elected from either the old or new member set. Partitions are unavailable to accept writes until the old and new sets reach joint consensus.
+
+Redpanda’s implementation of Raft includes changes to the reconfiguration protocol to ensure high availability even while group membership changes.
+
+== Redpanda’s approach to Raft reconfiguration
+
+Redpanda’s Raft implementation adds the following behavior when reconfiguring Raft groups:
+
+- In addition to leader and follower, a broker can also be in learner status. Learners are followers that cannot vote for a new leader, so they do not count towards quorum. Learners do not trigger leader elections and do not accept writes. New Raft group members start as learners. Brokers can be promoted or demoted between learner and voter. 
++
+This means that when new members are added to the Raft group member set, the group continues to operate without interruption since members that can form quorum still accept writes, even if the majority has technically changed due to the additional members. As soon as a learner is synced with the current leader, the learner is promoted to voter. Learners are promoted one at a time. 
+
+- Because all new nodes joining a Raft group start in learner status, the group does not yet have to enter the joint consensus phase as soon as new members join.
+
+- The Raft group only adds new members first. When all learners have been promoted to voters, the group can enter joint consensus. At this point, this is also a new group configuration that contains both the old as well as requested new member set. The current leader replicates this configuration to all brokers. 
++
+Only after learners are up to date with the leader will the group elect a new leader through joint consensus. Members that are not needed in the new configuration are demoted to learners before removal. The new leader then replicates the desired final configuration.
```

Comment:
This `Partitions are unavailable to accept writes until the old and new sets reach joint consensus.` can be removed